File: ocaml/difference-of-squares/difference_of_squares.ml
(* Square the sum of the numbers up to the given number
val square_of_sum : int -> int

(* Sum the squares of the numbers up to the given number *)
val sum_of_squares : int -> int

(* Subtract sum of squares from square of sums *)
val difference_of_squares : int -> int
*)

let pow n m = int_of_float (float_of_int n ** float_of_int m)

let range n = List.init n ((+) 1)

let sum = List.fold_left (+) 0

let square_of_sum n =
  pow (sum (range n)) 2

let sum_of_squares n =
  sum (List.map (fun x -> pow x 2) (range n))

let difference_of_squares n =
  square_of_sum n - sum_of_squares n

File: fsharp/difference-of-squares/DifferenceOfSquares.fs
module DifferenceOfSquares

let squareOfSum (number: int): int = 
    (fun x -> x*x)([1..number] |> List.sum)

let sumOfSquares (number: int): int = 
    [1..number] |> List.map (fun x -> x*x) |> List.sum

let differenceOfSquares (number: int): int = 
    squareOfSum number - sumOfSquares number